stumpy/scrump.py

```diff
@@ -835,7 +945,9 @@ def __init__(
                     indices,
                     s,
                     excl_zone,
-                ) = _preprocess_prescrump(T_A, m, s=s)
+                ) = _preprocess_prescrump(
+                    T_A, m, s=s, T_A_subseq_isconstant=T_A_subseq_isconstant
```

Comment:
In class `scrump`, when `pre_scrump==True`, we compute the prescrump.  

@seanlaw 
I think there is a code smell here but I am not sure. I would like to bring your attention to the following items:
* The input `T_A` must be original and not the preprocessed version `self._T_A`; because otherwise, the output `M_T` will not be correct and it will not have `np.inf` where it should have. One idea is to add the argument `T_subseq_isfinite` to the function `_preprocess_prescrump`.
* We pass `T_A_subseq_isconstant` and `T_B_subseq_isconstant`, and then we get `Q_subseq_isconstant` and `T_subseq_isconstant` in output. However, they are basically the same.

Reply:
> The input T_A must be original and not the preprocessed version self._T_A; because otherwise, the output M_T will not be correct and it will not have np.inf where it should have.

I'm not seeing the problem. Can you please elaborate? Currently, the code is correct, right?

> We pass T_A_subseq_isconstant and T_B_subseq_isconstant, and then we get Q_subseq_isconstant and T_subseq_isconstant in output. However, they are basically the same.

So, you think that `_preprocess_prescrump` should not pass these back? Or should we have changed `Q_subseq_isconstant` and `T_subseq_isconstant` to `T_A_subseq_isconstant` and `T_B_subseq_isconstant`?  What do you propose?

Reply:
> Currently, the code is correct, right?

Yes

> So, you think that `_preprocess_prescrump` should not pass these back? Or should we have changed `Q_subseq_isconstant` and `T_subseq_isconstant` to `T_A_subseq_isconstant` and `T_B_subseq_isconstant`? What do you propose?

I feel we are re-computing variables. So, I was thinking that we may do something like this (see code below) instead of calling the function `_preprocess_prescrump` and compute ALL required variables from scratch. 

```
if pre_scrump:
     _, σ_Q = core.compute_mean_
     _, Σ_T = core.compute_mean_std(T_B, self._m)
     indices = np.random.permutation(range(0, self._l, s)).astype(np.int64)
     P, I = _prescrump(...)
    
```
We just compute three more values and that's it. However, after implementing it locally, I got the following error:
```
SystemError: CPUDispatcher(<function _prescrump at 0x7fde740065f0>) returned a result with an exception set
```
Furthermore, I now think it is not as readable as the original version. 

So, maybe we should just leave it as it.

Reply:
> `Q_subseq_isconstant` and `T_subseq_isconstant` to `T_A_subseq_isconstant` and `T_B_subseq_isconstant`

I am going to push a commit to improve consistency in the code.